src/Microsoft.Diagnostics.Runtime/src/Common/ModuleInfo.cs

```diff
@@ -49,11 +51,22 @@ public class ModuleInfo
 
             try
             {
-                return _image = new PEImage(new ReadVirtualStream(_dataReader, (long)ImageBase, FileSize), isVirtual: true);
+                _image = new PEImage(new ReadVirtualStream(_dataReader, (long)ImageBase, FileSize), isVirtual: true);
+                _ = _image.IsManaged;
```

Comment:
What is the purpose of this line?

Reply:
The third commit is a demonstration to show that this is a layout issue. By calling `IsManaged`, the `PEReader` will read all headers and calculate the metadata location, and throw if the layout is incorrect.

Reply:
Ahh ok.